chromium/browser_type.go

```diff
@@ -387,11 +387,11 @@ func execute(
 		}()
 
 		if err := cmd.Wait(); err != nil {
-			logger.Errorf("BrowserType:execute", "cmd.Wait: %v", err)
-		}
-
-		if err := stdout.Close(); err != nil {
-			logger.Errorf("BrowserType:execute", "stdout.Close: %v", err)
+			if err.Error() != "signal: killed" {
+				logger.Errorf("BrowserType:execute", "cmd.Wait: %v", err)
+			} else {
+				logger.Debug("BrowserType:execute", "cmd.Wait: process killed")
+			}
```

Comment:
1. Yeah sure, how do you go about testing on Windows?
2. Yep, I was thinking the same tbh. I would prefer to create a new task to track that bug though as I don't think it'll be a quick fix, as I have a feeling that the context is being closed in k6 which is in turn killing the browser processor and not allowing us to gracefully shut it down.
  An approach could be to use a new context when creating the browser process; catch the context closing; and finally attempt to close browser process. But whatever approach we take will depend on https://github.com/grafana/k6/issues/2432 :/
3. Yep, I'll do that change with the log message.

Reply:
1. Discussed over Slack.

2. Sounds good about the new issue. [The context is cancelled after each iteration](https://github.com/grafana/k6/blob/be4fcb8cb4e30eebaa40d2c356e0be5cc1b21d2c/js/runner.go#L746-L748), besides the `RunContext` being cancelled at shutdown. So we should be able to stop the process gracefully [here](https://github.com/grafana/xk6-browser/blob/eb136726e37282fffad0c642062a04a6680a6adc/chromium/browser_type.go#L163), and ensure it works when running with multiple VUs/iterations, even though we can't ensure that it will work when k6 is stopped abruptly or at test end because of #2432.

Reply:
I've tested it on Windows and I get `exit status 1` from `err.Error()`. So I think it's:
1. Best to revert the logging for `cmd.Wait()` and move it into the new ticket where I will attempt to gracefully shutdown the browser, or otherwise log an error.
2. The `stdout.Close()` is definitely not needed in Windows either, so that can safely be deleted.

WDYT?

Reply:
New issue to try to gracefully shutdown the browser process is here: https://github.com/grafana/xk6-browser/issues/423

Reply:
Resolved in: 5222abb4d1d92ec43ce28fe9295f147fce1f61e2